code/modules/food_and_drinks/machinery/coffeemaker.dm

```diff
@@ -186,7 +200,66 @@
 		if(!user.transferItemToLoc(new_pot, src))
 			return TRUE
 		replace_pot(user, new_pot)
-		balloon_alert(user, "added pot")
+		update_appearance()
+		return TRUE //no afterattack
+
+	if (istype(attack_item, /obj/item/reagent_containers/cup/glass/coffee_cup) && !(attack_item.item_flags & ABSTRACT) && attack_item.is_open_container())
+		var/obj/item/reagent_containers/cup/glass/coffee_cup/new_cup = attack_item
+		if(new_cup.reagents.total_volume > 0 )
+			balloon_alert(user, "the cup must be empty!")
+			return TRUE
+		if(coffee_cups >= max_coffee_cups)
+			balloon_alert(user, "the cup holder is full!")
+			return TRUE
+		. = TRUE //no afterattack
+		if(!user.transferItemToLoc(attack_item, src))
+			return TRUE
+		coffee_cups++
+		update_appearance()
+		return TRUE //no afterattack
+
+	if (istype(attack_item, /obj/item/reagent_containers/condiment/pack/sugar))
+		var/obj/item/reagent_containers/condiment/pack/sugar/new_pack = attack_item
+		if(new_pack.reagents.total_volume < new_pack.reagents.maximum_volume)
+			balloon_alert(user, "the pack must be full!")
+			return TRUE
+		if(sugar_packs >= max_sugar_packs)
+			balloon_alert(user, "the sugar compartment is full!")
+			return TRUE
+		. = TRUE //no afterattack
+		if(!user.transferItemToLoc(attack_item, src))
+			return TRUE
+		sugar_packs++
+		update_appearance()
+		return TRUE //no afterattack
+
+	if (istype(attack_item, /obj/item/reagent_containers/condiment/creamer))
+		var/obj/item/reagent_containers/condiment/creamer/new_pack = attack_item
+		if(new_pack.reagents.total_volume < new_pack.reagents.maximum_volume)
+			balloon_alert(user, "the pack must be full!")
+			return TRUE
+		if(creamer_packs >= max_creamer_packs)
+			balloon_alert(user, "the creamer compartment is full!")
+			return TRUE
+		. = TRUE //no afterattack
+		if(!user.transferItemToLoc(attack_item, src))
+			return TRUE
+		creamer_packs++
+		update_appearance()
+		return TRUE //no afterattack
+
+	if (istype(attack_item, /obj/item/reagent_containers/condiment/pack/astrotame))
```

Comment:
These should all be else if

Reply:
Changed in [1758d26](https://github.com/tgstation/tgstation/pull/70991/commits/1758d260bf38349a463fb0c4c94b19bde1396d9f)